pkg/controller/ttl/ttlcontroller.go

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ttl
+
+import (
+	"strconv"
+	"sync/atomic"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/json"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	listers "k8s.io/kubernetes/pkg/client/legacylisters"
```

Comment:
I tried doing this. But there doesn't seem to be a way from go from:
"informers.NodeInformer"
(which is what we have in cmd/kuber-controller-manager/app/controllermanager.go"
to
the listers you want me to use.

I.e. from the NodeInformer all I can get is "cache.SharedIndexInformer" from which I don't know how to create the "NodeLister" from the generated lister...